### PR TITLE
howto writestate

### PR DESCRIPTION
**Pull request by willr3**
- revert to schema.json
- replace depricated methods and fix warnings
- add timeout for CI issues
- fix set-signal incrementing expected signal count and move docs
- How to upload files in a qDup script

### CODE REVIEW

File: docs/howto/writestate.adoc [hunk 1]
```diff
@@ -0,0 +1,56 @@
+= How to write state to separate files
+
+Complicated workloads, like running a suite of benchmarks or performance tests, create complicated output. That output can be parsed with `regex` into state values and included in the `run.json` that qDup creates or we write parts of the state to file as part of the qDup script to help separate the difference tests in the suite.
```
Comment (johnaohara) [inline docs/howto/writestate.adoc:3]:
```suggestion
Complicated workloads, like running a suite of benchmarks or performance tests, can create complicated output. That output can be parsed with `regex` into state variables and included in `run.json` that qDup creates with every run. Alternatively, we can write parts of the state to file(s) as part of the qDup script to help separate the different tests in the suite.
```

File: docs/howto/writestate.adoc [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+In our example we will use curl as our _performance test_ and our _benchmark suite_ will be a list of websites we want to measure. We will add `time` so we can measure the response time. The `sh` command is rather straight forward
```
Comment (johnaohara) [inline docs/howto/writestate.adoc:4]:
```suggestion

In our example we will use curl as our _performance test_ and our _benchmark suite_ will be a list of websites we want to measure. We will add `time` so we can measure the response time. The `sh` command is rather straight forward
```
Exra space to split up the paragraphs

File: docs/howto/writestate.adoc [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    - regex: "HTTP/\\d\\.\\d (?<code>\\d{3})"
+    - sh: echo '{"real":"${{real}}","user":"${{user}}","sys":"${{sys}}","code":"${{code}}"} > "${{target}}.json"
+    - queue-download: "${{target}}.json"
+```
+The above example constructs the json in the call to echo but we could also separate that into a `set-state` or `js` command.
+```yaml
+- set-state: toWrite {"real":"${{real}}","user":"${{user}}","sys":"${{sys}}","code":"${{code}}"}
+  then:
+  - sh : echo '${{toWrite}}' > "${{target}}.json"
+```
```
Comment (johnaohara) [inline docs/howto/writestate.adoc:39]: i wasn't thinking about manually constructing the JSON output here, but using RUN state variable to create it for us, e.g.;

```yaml
  - sh: time curl -v "${{target}}"
    then:
    - regex: "real\\s+(?<real>\\S+)"
      then:
        - set-state: RUN.results.${{target}}.real ${{real}}
    - regex: "user\\s+(?<user>\\S+)"
      then:
        - set-state: RUN.results.${{target}}.user ${{user}}
    - regex: "sys\\s+(?<sys>\\S+)"
      then:
        - set-state: RUN.results.${{target}}.sys ${{sys}}
    - regex: "HTTP/\\d\\.\\d (?<code>\\d{3})"
      then:
        - set-state: RUN.results.${{target}}.code ${{code}}
  - sh: echo '${{RUN.results}}' > metrics.json
  - queue-download: metrics.json
```

**Comment by johnaohara** [comment]
@willr3 as you mentioned, https://github.com/RedHatPerf/project-tracking/issues/300 could be covered by this guide if we dynamically add state variables

File: docs/howto/writestate.adoc (hunk 3 above)
Comment (willr3) [reply docs/howto/writestate.adoc:51]: I added an example of adding each regex to a `toWrite` but I left off the `RUN.` prefix to introduce that in a different section

**Pull request merged**